ostd/src/task/processor.rs

```diff
@@ -99,23 +80,34 @@ fn switch_to_task(next_task: Arc<Task>) {
         );
     }
 
-    let current_task_ctx_ptr = match current_task() {
-        None => get_idle_task_ctx_ptr(),
-        Some(current_task) => {
-            let ctx_ptr = current_task.ctx().get();
+    let irq_guard = crate::trap::disable_local();
 
-            let mut task_inner = current_task.inner_exclusive_access();
+    let current_task_ptr = CURRENT_TASK_PTR.load();
 
-            debug_assert_ne!(task_inner.task_status, TaskStatus::Sleeping);
-            if task_inner.task_status == TaskStatus::Runnable {
-                drop(task_inner);
-                GLOBAL_SCHEDULER.lock_irq_disabled().enqueue(current_task);
-            } else if task_inner.task_status == TaskStatus::Sleepy {
-                task_inner.task_status = TaskStatus::Sleeping;
-            }
+    let current_task_ctx_ptr = if current_task_ptr.is_null() {
+        // SAFETY: Interrupts are disabled, so the pointer is safe to be fetched.
+        unsafe { BOOTSTRAP_CONTEXT.get() }
+    } else {
+        // SAFETY: The pointer is not NULL and set as the current task.
+        let cur_task_arc = unsafe {
+            let restored = Arc::from_raw(current_task_ptr);
+            let _ = core::mem::ManuallyDrop::new(restored.clone());
+            restored
```

Comment:
These lines are duplicated with `current_task`; so is the earlier `if current_task_ptr.is_null()`. I think the parent if-else statement can be simplified by using `current_task`.

```rust
let current_task_ctx_ptr = if let Some(task) = current_task() {
    ...
} else {
    ...
}
```

Reply:
NO. I intentionally duplicate these lines because `CpuLocalCell::read()` is a black box and the compiler must generate a memory read instruction for it. It won't be reordered or optimized away. My version saves one `CpuLocalCell::read()` operation thus ensuring better performance.